contracts/zaps/ZapCrvErc20Generic.sol

```diff
@@ -0,0 +1,299 @@
+        // if there are more than 1 pools, assume the second one is a 3pool
+        if (_data.pools.length == 2) {
+            // build the funding array
+            uint256[3] memory positions;
+            positions[uint256(_data.indices[1])] = _data.amount;
+            // the new amount will be the 3pool lp tokens received
+            amount = _data.pools[1].add_liquidity(positions, 0);
+        }
+
+        // fund the base pool to get the target LP
+        uint256[2] memory positions;
+        positions[_data.indices[0]] = amount;
+
+        // amount is now the targetLp value received
+        amount = _data.pools[0].add_liquidity(positions, 0);
+
+        // transfer target lp to the tranche and call prefundedDeposit
+        ITranche tranche = _deriveTranche(address(_position), _expiration);
+        _data.targetLp.transfer(address(_position), amount);
+
+        (uint256 ptMinted, uint256 ytMinted) = tranche.prefundedDeposit(
+            msg.sender
+        );
+
+        require(ytMinted >= amount, "Not enough YT minted");
+        require(ptMinted >= _ptExpected, "Not enough PT minted");
+        return (ptMinted, ytMinted);
+    }
+
+    /// @notice This function takes principal and/or yield tokens and redeems them for an output token.
+    /// @param _expiration Expiration of the target tranche.
+    /// @param _position The wrapped position contract address.
+    /// @param _amountPt Amount of principal tokens to redeem.
+    /// @param _amountYt Amount of yield tokens to redeem.
+    /// @param _outputExpected The minimum amount of principal tokens to mint.
+    function zapOut(
+        uint256 _expiration,
+        address _position,
+        uint256 _amountPt,
+        uint256 _amountYt,
+        uint256 _outputExpected,
+        ZapOut memory _data,
+        PermitData[] calldata _permitCallData
+    ) external reentrancyGuard notFrozen preApproval(_permitCallData) {
+        ITranche tranche = _deriveTranche(address(_position), _expiration);
+        IERC20 yt;
+        uint256 balance;
+        // withdraw appropriate amount of lp to this contract
+        if (_amountPt > 0) {
+            tranche.transferFrom(msg.sender, address(this), _amountPt);
+            balance += tranche.withdrawPrincipal(_amountPt, address(this));
+        }
+        if (_amountYt > 0) {
+            yt = IERC20(tranche.interestToken());
+            yt.transferFrom(msg.sender, address(this), _amountYt);
+            balance += tranche.withdrawInterest(_amountYt, address(this));
+        }
+
+        require(balance > 0, "no balance");
+
+        // balance is the output of the liquidity removal.
+        balance = _data.pools[0].remove_liquidity_one_coin(
+            balance,
+            _data.indices[0],
+            0
+        );
+
+        // if there are more than 1 pools, assume the second one is a 3pool.
+        // if the output token is a 3pool lp, balance tracks the target token output amount
+        if (_data.pools.length > 1) {
```

Comment:
In ZapIn your conditional is `if (_data.pools.length == 2)` ... wouldn't it be better to be consistent?